services/wallet/transfer/commands_sequential.go

```diff
@@ -1318,3 +1305,42 @@ func nextRange(maxRangeSize int, prevFrom, zeroBlockNumber *big.Int) (*big.Int,
 
 	return from, to
 }
+
+func accountLimiterTag(account common.Address) string {
+	return transferHistoryTag + "_" + account.String()
+}
+
+func createChainClientWithLimiter(client chain.ClientInterface, account common.Address, limiter chain.RequestLimiter) (chain.ClientInterface, error) {
+	// Each account has its own limit and a global limit for all accounts
+	accountTag := accountLimiterTag(account)
+	chainClient := chain.ClientWithTag(client, accountTag, transferHistoryTag)
+
+	// Check if limit is already reached, then skip the comamnd
+	if allow, err := limiter.Allow(accountTag); !allow {
+		log.Info("fetchHistoryBlocksForAccount limit reached", "account", account, "chain", chainClient.NetworkID(), "error", err)
+		return nil, err
+	}
+
+	if allow, err := limiter.Allow(transferHistoryTag); !allow {
+		log.Info("fetchHistoryBlocksForAccount common limit reached", "chain", chainClient.NetworkID(), "error", err)
+		return nil, err
+	}
+
+	limit, _ := limiter.GetLimit(accountTag)
+	if limit == nil {
+		err := limiter.SetLimit(accountTag, transferHistoryLimitPerAccount, chain.LimitInfinitely)
+		if err != nil {
+			log.Error("fetchHistoryBlocksForAccount SetLimit", "error", err, "accountTag", accountTag)
```

Comment:
Why we don't return on error? I mean, if saving to the storage fails isn't that an error worth returning, cause no point in having a client with limits anymore?